Task 4/Task 4/Task 4/FileEventsInfo.cs

```diff
@@ -30,34 +31,29 @@ public FileEventsInfo(string path, string oldPath, DateTime changes, string type
         public DateTime LastChangesTime { get; private set; }
 
         [JsonInclude]
-        public string EventType { get; private set; }
+        public FileActions EventType { get; private set; }
 
         [JsonInclude]
         public string Content { get; private set; }
 
         public void PrintState()
         {
-            Print.PrintMessage($"FullPath: {FullPath}" + Environment.NewLine +
-                                   $"Time Of Changes: {LastChangesTime} " + Environment.NewLine +
-                                   $"Type Of Changes: {EventType} " + Environment.NewLine +
-                                   $"Content: {Content}" + Environment.NewLine);
+            Console.WriteLine($"FullPath: {FullPath}" + Environment.NewLine +
+                              $"Time Of Changes: {LastChangesTime} " + Environment.NewLine +
+                              $"Type Of Changes: {EventType} " + Environment.NewLine);
         }
 
-        private string CheckEmptyOrNull (string str)
+        private string CheckFullPath (string str)
         {
-            if (str == String.Empty || str == null)
-            {
-                throw new ArgumentException("You can't put empty string or null here.");
-            }
+            if (String.IsNullOrWhiteSpace(str))
+                throw new ArgumentException(String.Format("You can't put null or white spaces into {0}", FullPath));
```

Comment:
```cs
throw new ArgumentException($"You can't put null or white spaces into {FullPath}");
```
This is more readable and also in this case converts to concatenation on compile (faster than Format).